src/org/jitsi/impl/neomedia/RTPConnectorInputStream.java

```diff
@@ -743,9 +743,9 @@ private void runInReceiveThread()
 
         while (!closed)
         {
-            // http://code.google.com/p/android/issues/detail?id=24765
-            if (OSUtils.IS_ANDROID)
-                p.setLength(PACKET_RECEIVE_BUFFER_LENGTH);
+            // Reset the buffer, because the previous call to receive() might
+            // have bumped the offset or even changed the byte[].
+            p.setData(buffer, 0, buffer.length);
```

Comment:
For Octo I override receive(), and I strip some headers from the packet by bumping the offset by 8 bytes. Without this reset the next call to receive uses offset=8, the underlying socket receives at that offset, then I strip 8 bytes and offset grow to 16, etc, etc.